src/renderer/model/Texture.cpp

```diff
@@ -0,0 +1,54 @@
+#include <rg/renderer/model/Texture.hpp>
+
+#include <glad/glad.h>
+#include <spdlog/spdlog.h>
+#include <stb/stb_image.h>
+
+namespace rg {
+
+Texture::Texture(unsigned int id, TextureType type)
+        : texture_id{id}, type{type} {
+}
+
+Texture::Texture(const std::string& path, TextureType type)
+        : texture_id{0}, type{type} {
+    glGenTextures(1, &texture_id);
+    glBindTexture(GL_TEXTURE_2D, texture_id);
+
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER,
+                    GL_LINEAR_MIPMAP_LINEAR);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+
+    int width, height, num_channels, comp;
```

Comment:
ne znam šta bi trebalo comp da bude ali je unused

Reply:
Ovaj commit bi trebalo da sređuje to:
https://github.com/pearpanda/computer-graphics-project/pull/8/commits/6a5b6d586a983762418139888936460a4eb63387